pkg/channel/message_dispatcher.go

```diff
@@ -104,29 +115,28 @@ func (d *MessageDispatcherImpl) DispatchMessageWithRetries(ctx context.Context,
 	// Otherwise, they are filled with the original message
 	var responseMessage cloudevents.Message
 	var responseAdditionalHeaders nethttp.Header
-
+	var dispatchExecutionInfo *DispatchExecutionInfo
 	if destination != nil {
 		var err error
 		// Try to send to destination
 		messagesToFinish = append(messagesToFinish, message)
 
-		ctx, responseMessage, responseAdditionalHeaders, err = d.executeRequest(ctx, destination, message, additionalHeaders, retriesConfig)
+		ctx, responseMessage, responseAdditionalHeaders, dispatchExecutionInfo, err = d.executeRequest(ctx, destination, message, additionalHeaders, retriesConfig)
 		if err != nil {
 			// DeadLetter is configured, send the message to it
 			if deadLetter != nil {
-
-				_, deadLetterResponse, _, deadLetterErr := d.executeRequest(ctx, deadLetter, message, additionalHeaders, retriesConfig)
+				_, deadLetterResponse, _, dispatchExecutionInfo, deadLetterErr := d.executeRequest(ctx, deadLetter, message, additionalHeaders, retriesConfig)
 				if deadLetterErr != nil {
-					return fmt.Errorf("unable to complete request to either %s (%v) or %s (%v)", destination, err, deadLetter, deadLetterErr)
+					return dispatchExecutionInfo, fmt.Errorf("unable to complete request to either %s (%v) or %s (%v)", destination, err, deadLetter, deadLetterErr)
```

Comment:
nit but just to make sure I follow along here, it seems like it's not 'either' isn't it both? at this point the original request failed (correct?) as well as executeRequest to deadletter failed?

Reply:
Yes you are right, a few lines bellow the code uses `and` so I could update the message, although the PR does not touch this part.

Reply:
No worries, we can clean it up in a follow on PR, I just wanted to make sure that my mental model was correct and if not, that I could gain understanding 👍